datajoint/declare.py

```diff
@@ -414,6 +414,9 @@ def compile_attribute(line, in_key, foreign_key_sql):
     match['comment'] = match['comment'].replace('"', '\\"')   # escape double quotes in comment
     category, type_match = match_type(match['type'])
 
+    if match['comment'].startswith(':'):
+        raise DataJointError('An attribute comment must not start with a colon in comment "{comment}"'.format(**match))
```

Comment:
isn't the problem colon anywhere in the comment?

Reply:
Based on the new test case, it is OK to actually have `:` in the comment, but that's also why I'm not sure when it is that this gets triggered and why.

Reply:
Actually, I vaguely recall this having to do with the custom DataJoint datatype handling that uses tag of form `:external:` at the very beginning of the comment.

Reply:
Custom datatypes are encoded as `:custom-type: user comment` If it does not begin with a `:`, then it's fine.

Reply:
@ixcat, The actual fix was to fix the regexp that was previously greedily matching. The check above is fixing a separate problem from what you reported in #633

Reply:
Makes sense & Starting a comment with a colon would be an odd thing to do.. will change request / merge